### DISCUSSION TITLE
Sidebars do not follow the content when long

### DISCUSSION THREAD

**Discussion by manticore-projects** [pradyunsg/furo#667] [Ideas]
Greetings.

Furo is great and we love it. The right side bar is especially useful for API documentation.
But there are two big caveats:

1) it does not follow, when the TOC is long and the page scrolls (which is very usual for API)
2) there is not simple search (which again was useful for API).

Please have a look at https://jsqlparser.github.io/JSqlParser/javadoc_snapshot.html#numericbind
It illustrates the challenges and also overlays a JavaScript generated SideBar which solves those 2 caveats better. Maybe you can integrate that directly.

**Comment by pradyunsg** [comment]
Yea, that's a thing I had struggled with a bunch before giving up. Browsers are weird. 😅

I'll welcome fixes for this behaviour!

**Comment by pradyunsg** [comment]
I don't think sidebar search is something I'm going to be implementing -- that feels particularly niche but having the content show up in frame is something that's been attempted before as well as been requested before.

**Comment by manticore-projects** [comment]
Greetings.

That is very ok of course, since I am happy with this my niche solution.
I only wanted to point out that the current right sidebar for an API has some limitations (and I will need to switch it off for any API related site).

Thank you for your great work and effort. It is much appreciated.
